### PR TITLE
AEA-1158 Fix docs and protocol terminal states

### PR DESCRIPTION
**Pull request by DavidMinarsch**
## Proposed changes

This PR fixes an issue in docs and adds proper terminal states to all protocols.

## Fixes

-

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

-

### CODE REVIEW

File: aea/protocols/generator/validate.py [hunk 1]
```diff
@@ -685,6 +696,39 @@ def _validate_termination(
                 ),
             )
 
+    # check that there are no repetitive performatives in termination
+    duplicates_dict: Dict[str, int] = {}
+    for performative in termination:
+        duplicates_dict[performative] = duplicates_dict.get(performative, 0) + 1
+    duplicates = [k for k, v in duplicates_dict.items() if v > 1]
```
Comment (DavidMinarsch) [inline aea/protocols/generator/validate.py:703]: just use `no_duplicates = len(termination) == len(list(set(termination)))`, this is very non-pythonic

File: aea/protocols/generator/validate.py [hunk 2]
```diff
@@ -823,6 +871,7 @@ def _validate_dialogue_section(
         ) = _validate_termination(
             cast(List[str], protocol_specification.dialogue_config["termination"]),
             performatives_set,
+            cast(Set[str], terminal_performatives_from_reply),
```
Comment (DavidMinarsch) [inline aea/protocols/generator/validate.py:874]: Don't cast here; just treat optional type properly in the method...

File: aea/protocols/generator/validate.py (hunk 1 above)
Comment (5A11) [reply aea/protocols/generator/validate.py:703]: Yea, I did something similar initially, but this gives a better error message, by printing what performatives are being repeated. 

If you think it is not that useful I simply the code.

File: aea/protocols/generator/validate.py (hunk 2 above)
Comment (5A11) [reply aea/protocols/generator/validate.py:874]: We know it won't be Optional (None). If it was, the previous `# Validate reply` block would return False and we won't reach this point. 

I moved cast at the beginning of the block for more clarity.

**Pull request merged**